lib/src/extensions.dart

```diff
@@ -186,6 +186,9 @@ extension DartTypeExtension on DartType? {
         definitions.any((d) => i.isSameAs(d.name, d.library));
 
     var self = this;
+    if (self is TypeParameterType) {
+      self = self.typeForInterfaceCheck;
```

Comment:
I'd prefer that we only use `self` as a synonym for `this` (and then only for the purposes of improving type promotion). If we're going to be setting the variable to something other than `this`, let's rename it. Consider `typeToCheck`.

Reply:
Thanks, done.